src/neo/Ledger/TransactionVerificationContext.cs

```diff
@@ -25,13 +36,29 @@ public bool CheckTransaction(Transaction tx, StoreView snapshot)
         {
             BigInteger balance = NativeContract.GAS.BalanceOf(snapshot, tx.Sender);
             senderFee.TryGetValue(tx.Sender, out var totalSenderFeeFromPool);
+
             BigInteger fee = tx.SystemFee + tx.NetworkFee + totalSenderFeeFromPool;
-            return balance >= fee;
+            if (balance < fee) return false;
+
+            var oracle = tx.GetAttribute<OracleResponse>();
+            if (oracle != null &&
+                (!oracleResponses.TryGetValue(oracle.Id, out var hash) || hash != tx.Hash))
```

Comment:
```suggestion
                (oracleResponses.Contains(oracle.Id)))
```

Reply:
We need to know that is not this tx, in order to to ensure that it's a duplicate

Reply:
Duplicate transactions should not be verified. The verification of the fee is the same.

Reply:
Duplicate transactions should be avoided, which may lead to insufficient transaction costs for subsequent responses.

Reply:
Done!